### ISSUE TITLE
Backwards-incompatible change in cf curl behavior

### ISSUE THREAD

**Issue by mdelillo** [cloudfoundry/cli#788]
This change is causing acceptance test suites ([cf-acceptance-tests](https://github.com/cloudfoundry/cf-acceptance-tests), [cf-redis-smoke-tests](https://github.com/pivotal-cf/cf-redis-smoke-tests), [cf-rabbitmq-smoke-tests](https://github.com/pivotal-cf/cf-rabbitmq-smoke-tests)) to fail. Empty input to the `-d` flag should be ignored, as it was in versions 6.15.0 and before. It looks like the breaking code was added [here](https://github.com/cloudfoundry/cli/commit/b2ca2d045f79f260339798ae3a4878f1ed31d1e8#diff-a8ab279fb494b4f93471cd4595c6b4caR80).

### Command

```
cf curl /v2/info -d ""
```

### CLI Version

6.16.0, 6.16.1

### Error

```
FAILED
invalid input:
```

### CF Trace

```
$ CF_TRACE=true cf curl /v2/info -d ""
FAILED
invalid input:
FAILED
invalid input:
```

### Platform Details

Mac OS X 10.10.3

### Shell

iTerm

Thanks,
@mdelillo @sclevine

**Comment by dkoper** [comment]
Thanks @mdelillo, @sclevine, we're looking into it.

**Comment by davidwadden** [comment]
@dkoper,

it looks like a [fix](https://github.com/cloudfoundry/cli/commit/f59c7bb27eac542426cbb4b66aa51c7081f5cfbc) has already been checked in.  can you guys cut a patch release with this fix ASAP?  

both v6.16.0 and v6.16.1 break CATs as well as other acceptance test suites.

thanks,
@davidwadden

**Comment by dkoper** [comment]
Hi @davidwadden!

It sounds the CATs are relying on something `cf curl` is not meant to support?
`-d            HTTP data to include in the request body`

If you want it to be ignored, don't set the flag?
We're fixing it, as `curl` seems to accept it too, but curl also then changes the request to a POST, which may not be what you want?

**Comment by davidwadden** [comment]
As we mentioned, the fix is already in -- https://github.com/cloudfoundry/cli/commit/f59c7bb27eac542426cbb4b66aa51c7081f5cfbc

We need a release of the CF CLI with this fix in order to use 6.16+ with several acceptance test suites for CF, including [CATs](https://github.com/cloudfoundry/cf-acceptance-tests).  If you think the behavior is mis-specified, feel free to work with the appropriate teams to fix in that in a future release, but the current version of the CF CLI is not compatible with current versions of CATs.

**Comment by sclevine** [comment]
@dkoper Are there any plans to cut a patch release with f59c7bb27eac542426cbb4b66aa51c7081f5cfbc soon? It seems that cf-test-helpers adapted to this regression, but a number of acceptance test suites aren't using the most recent cf-test-helpers (such as cf-mysql-release).

**Comment by dkoper** [comment]
Hi @sclevine, all,

We want to fix this properly and have placed this follow-up story at the top of our backlog:
https://www.pivotaltracker.com/story/show/115380315

I'd like to work with the relevant teams to understand how `cf curl` is used currently and how it should be used. We haven't encountered this issue in our runs of CATS, and haven't heard back from a request to the release-integration team. I suspect we run only a subset.

**Comment by dkoper** [comment]
Apologies for the late follow-up. I reached out to @sclevine and got more details.
The curl issue has been fixed in Edge, will be included in the next release.
In line with `curl`, inclusion of `-d` means POST unless overridden by `-X`.
In the meantime, you can either revert to the previous cf CLI version, not use `-d ""`, or wait for the next release and make sure you add `-X GET` if you don't want a POST.

**Comment by dkoper** [comment]
This has been fixed in cf CLI v6.17.0.
https://github.com/cloudfoundry/cli#downloads

Thanks again!
Dies Koper
CF CLI PM
